### PR TITLE
Delete method for PersistentVolumeClaim

### PR DESCRIPTION
**Pull request by yprokule**
Methods to delete _PersistentVolumeClaim_ resources from the cluster

### CODE REVIEW

File: pkg/storage/pvc.go [hunk 1]
```diff
@@ -173,6 +173,43 @@ func (builder *PVCBuilder) Create() (*PVCBuilder, error) {
 	return builder, nil
 }
 
+// Delete removes PVC from cluster.
+func (builder *PVCBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		glog.V(100).Infof("persistentVolumeClaim %s is invalid- %v", builder.Definition.Name, err)
```
Comment (achuzhoy) [inline pkg/storage/pvc.go:179]: I'd suggest instead of "invalid-" to use "invalid:"

**Comment by achuzhoy** [comment]
do you want to add deleteandwait too?

**Review by cdvultur — COMMENTED** [review]
I would use PersistentVolumeClaim instead persistentVolumeClaim/PVC

File: pkg/storage/pvc.go (hunk 1 above)
Comment (yprokule) [reply pkg/storage/pvc.go]: updated

[the remark this answers is not in the document]
Comment (yprokule) [reply pkg/storage/pvc.go]: updated

File: pkg/storage/pvc.go [hunk 2]
```diff
@@ -173,6 +173,43 @@ func (builder *PVCBuilder) Create() (*PVCBuilder, error) {
...
+
+		return err
+	}
+
+	glog.V(100).Infof("Delete persistentVolumeClaim %s from %s namespace",
```
[the remark this answers is not in the document]
Comment (yprokule) [reply pkg/storage/pvc.go]: updated

File: pkg/storage/pvc.go [hunk 3]
```diff
@@ -173,6 +173,43 @@ func (builder *PVCBuilder) Create() (*PVCBuilder, error) {
...
+		return nil
+	}
+
+	err := builder.apiClient.PersistentVolumeClaims(builder.Definition.Namespace).Delete(
+		context.TODO(), builder.Definition.Name, metaV1.DeleteOptions{})
+
+	if err != nil {
+		glog.V(100).Infof("Failed to delete PersistentVolumeClaim %s from %s namespace",
+			builder.Definition.Name, builder.Definition.Namespace)
+		glog.V(100).Infof("PVC deletion error: %v", err)
```
[the remark this answers is not in the document]
Comment (yprokule) [reply pkg/storage/pvc.go]: updated

**Comment by yprokule** [comment]
> do you want to add deleteandwait too?

@achuzhoy sure, no biggie. Added `DeleteAndWait` method as well

**Review by cdvultur — APPROVED** [review]
/lgtm

File: pkg/storage/pvc.go [hunk 4]
```diff
@@ -173,6 +175,75 @@ func (builder *PVCBuilder) Create() (*PVCBuilder, error) {
...
+
+	builder.Object = nil
+
+	return err
+}
+
+// DeleteAndWait deletes PersistentVolumeClaim and waits until it's removed from the cluster.
+func (builder *PVCBuilder) DeleteAndWait(timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		glog.V(100).Infof("PersistentVolumeClaim %s is invalid: %v", builder.Definition.Name, err)
```
Comment (achuzhoy) [inline pkg/storage/pvc.go:218]: missing namespace in the glog

File: pkg/storage/pvc.go [hunk 5]
```diff
@@ -173,6 +175,75 @@ func (builder *PVCBuilder) Create() (*PVCBuilder, error) {
...
+
+		return err
+	}
+
+	glog.V(100).Infof("Deleting PersistenVolumeClaim %s and waiting for the removal to complete",
```
Comment (achuzhoy) [inline pkg/storage/pvc.go:223]: missing namespace in the glog

File: pkg/storage/pvc.go (hunk 4 above)
Comment (yprokule) [reply pkg/storage/pvc.go]: thanks, updated

**Review by achuzhoy — APPROVED** [review]

**Review by cdvultur — APPROVED** [review]
/lgtm

**Pull request merged**